lib/ansible/modules/cloud/amazon/aws_waf_rule.py

```diff
@@ -27,15 +27,23 @@
         required: yes
     metric_name:
         description:
-        - A friendly name or description for the metrics for the rule
-        - The name can contain only alphanumeric characters (A-Z, a-z, 0-9); the name can't contain whitespace.
-        - You can't change metric_name after you create the rule
-        - Defaults to the same as name with disallowed characters removed
+          - A friendly name or description for the metrics for the rule
+          - The name can contain only alphanumeric characters (A-Z, a-z, 0-9); the name can't contain whitespace.
+          - You can't change metric_name after you create the rule
+          - Defaults to the same as name with disallowed characters removed
+    type:
+        description:
+          - 'regular' or 'rate_based'
```

Comment:
I believe the use of quotes here is confusing the YAML parser. Also, you can document the `choices`
```suggestion
          - Which rule set to use.
        choices:
            - regular
            - rate_based
          
```

Reply:
Thanks for the feedback, I will try to fix that.
Also, I think I would need to push a fix to the `aws_waf_facts` module, because it dosesn't work either with rate-based rules :(